### DISCUSSION TITLE
Listbox Widget Addition

### DISCUSSION THREAD

**Discussion by srturmelle** [TomSchimansky/CustomTkinter#617] [General]
Hello.  I'd like to thank you all for this great module.  We've recently started generating some automation scripts at work and this has become our "go-to" for front-end gui implementation.

Is there a plan to implement the listbox widget into the module?

**Comment by TomSchimansky** [comment]
Hi, nice to hear you like the library. I have the plan to implement some kind of listbox or table in the future. But I cannot say when it will be ready, cause I'm studying right now and often there is not that much time left for working on this project...

**Reply by Banta2000** [reply]
Listbox would be awesome! Thank you so much for your work.

**Reply by ghost** [reply]
> Hi, nice to hear you like the library. I have the plan to implement some kind of listbox or table in the future. But I cannot say when it will be ready, cause I'm studying right now and often there is not that much time left for working on this project...

hello sir I am Aryan and I am working on a project, Listbox is needed for that project to continue. but Listbox was not in this library. So, I needed to stop that project midway and use this module in any other project. but yes I want to comment that the texture and the new features added to Tkinter are awesome. Your code is so good and the documentary is easy to learn and explore. Thank you.

**Comment by frank572** [comment]
I agree, your work is amazing...thank you so much! I would also love to see a Listbox, preferably with an optional scroll bar (such as you implemented with the text box). I use Listboxes in many applications - e.g. a list of movies to select, a list of songs to play. They are indispensable. In the meantime, I've fudged together a listbox using a tkinter Listbox that sits inside a CTk frame, assigning the Roboto typeface to the Listbox. Unfortunately, I'm still stuck with the clunky scrollbar. When you have a chance, would be much appreciated!

**Reply by roelvanderzee** [reply]
Hi Tom, your work is greatly appreciated. It's the best. A Listbox, indeed preferably with an optional scroll bar as Frank572 mentions, are of great help.

**Comment by talatozmusluman** [comment]
Any progress on Listbox ? That would help us a lot!

**Comment by BloomAutist47** [comment]
any updates?

**Reply by Akascape** [reply]
No progress on this one. But it's easy to make that widget using scrollable frame and buttons.

**Comment by frank572** [comment]
You can use a tk listbox inside a ctk frame. Works perfectly well. It has no scrollbar, of course, but the user can scroll with the mouse or up/down arrow keys. In addition, the Roboto font used by ctk is available to all widgets (tk or ctk) and so you can likewise maintain consistency by using Roboto in the listbox. Until the 'real thing' comes along, this is a perfectly useable substitute.

**Comment by Akascape** [comment]
@srturmelle @frank572 @BloomAutist47 @talatozmusluman 

**Hi, I just made this quick listbox for customtkinter:**
https://github.com/Akascape/CTkListbox

![](https://user-images.githubusercontent.com/89206401/247524994-256c135f-3d6c-4daa-87a5-b4572a2c905f.jpg)

**Reply by BloomAutist47** [reply]
Thanks a lot mate. Exactly what I needed!

**Reply by avalon60** [reply]
Hey, @Akascape, does that do multiple selections?

**Reply by Akascape** [reply]
@avalon60 I have added multiple selection. Use `multiple_selection=True`. It will return a list of selected options.

**Reply by avalon60** [reply]
Fantastic!
